pkg/pillar/cmd/zedrouter/acl.go

```diff
@@ -770,141 +800,235 @@ func diffIpsets(newIpsets, oldIpsets []string) ([]string, []string, bool) {
+	valueList := make([]string, len(matchTypes))
+	valueList1 := make([]string, len(matchTypes))
 
-	ipVer := determineIpVer(isMgmt, bridgeIP)
-	rules, err := aclToRules(bridgeName, vifName, ACLs, ipVer,
-		bridgeIP, appIP)
-	if err != nil {
-		return err
-	}
-	dropRules, err := aclDropRules(bridgeName, vifName)
-	if err != nil {
-		return err
-	}
-	rules = append(rules, dropRules...)
-	for _, rule := range rules {
-		log.Debugf("deleteACLConfiglet: rule %v\n", rule)
-		args := rulePrefix("-D", isMgmt, ipVer, vifName, appIP, rule)
-		if args == nil {
-			log.Debugf("deleteACLConfiglet: skipping rule %v\n",
-				rule)
-			continue
+	for idx, matchType := range matchTypes {
+		for _, match := range ace.Matches {
+			if matchType == match.Type {
+				valueList[idx] = match.Value
+			}
 		}
-		args = append(args, rule...)
-		if ipVer == 4 {
-			err = iptables.IptableCmd(args...)
-		} else if ipVer == 6 {
-			err = iptables.Ip6tableCmd(args...)
-		} else {
-			err = errors.New(fmt.Sprintf("ACL: Unknown IP version %d", ipVer))
+		for _, match := range ace1.Matches {
+			if matchType == match.Type {
+				valueList1[idx] = match.Value
+			}
 		}
-		if err != nil {
-			return err
+	}
+	for idx, value := range valueList {
+		value1 := valueList1[idx]
+		if value == "" || value1 == "" ||
+			value != value1 {
+			return false
 		}
 	}
-	if !isMgmt {
-		// Remove mangle rules for IPv6 packets added above
-		// XXX error checks?
-		iptables.Ip6tableCmd("-t", "mangle", "-D", "PREROUTING", "-i", bridgeName,
-			"-p", "tcp", "-j", "CHECKSUM", "--checksum-fill")
-		iptables.Ip6tableCmd("-t", "mangle", "-D", "PREROUTING", "-i", bridgeName,
-			"-p", "udp", "-j", "CHECKSUM", "--checksum-fill")
+	return true
+}
+
+// utility routines for IpTables Rules
+func executeIPTablesRule(operation string, rule types.IPTablesRule) error {
+	var err error
+	ruleStr := []string{}
+	if rule.Table != "" {
+		ruleStr = append(ruleStr, "-t")
+		ruleStr = append(ruleStr, rule.Table)
+	}
+	ruleStr = append(ruleStr, operation)
+	ruleStr = append(ruleStr, rule.Chain)
+	ruleStr = append(ruleStr, rule.Prefix...)
+	ruleStr = append(ruleStr, rule.Rule...)
+	if rule.IPVer == 4 {
+		err = iptables.IptableCmd(ruleStr...)
+	} else if rule.IPVer == 6 {
+		err = iptables.Ip6tableCmd(ruleStr...)
+	} else {
+		errStr := fmt.Sprintf("ACL: Unknown IP version %d", rule.IPVer)
+		err = errors.New(errStr)
 	}
-	// XXX see above
-	if false && ipVer == 6 && !isMgmt {
-		// Manually delete the manual add above
-		// XXX error checks?
-		iptables.Ip6tableCmd("-D", "FORWARD", "-i", bridgeName, "-o", "dbo1x0",
-			"-j", "DROP")
+	return err
+}
+
+// handle network instance level ACL rules
+// Network Instance Level ACL rule handling routines
+func handleNetworkInstanceACLConfiglet(op string, aclArgs types.AppNetworkACLArgs) error {
+
+	log.Infof("bridge(%s, %v) iptables op: %v\n", aclArgs.BridgeName, aclArgs.BridgeIP, op)
+	rulesList := networkInstanceBridgeRules(aclArgs)
+	for _, rule := range rulesList {
```

Comment:
Please add comment about inserting in order (to ensure that log occurs before the associated drop).